### ISSUE TITLE
Accessibility Improvement - GLPI 11.x  there are a lot of buttons and form elements without aria labels or titles

### ISSUE THREAD

**Issue by citygabor** [glpi-project/glpi#22034]
Hello GLPI Community and GLPI Engineering Team,

I would like to report some accessibility issues to you about the new GLPI 11.x version.
I  have a visual loss and working with screen reader for my daily computer usage . I am mainly working with JAWS, but sometimes I validate the applications with Narrator (Windows built in screen reader) and NVDA (Open source and freely available screen reader) as well.
Please  note I have collected some of my findings with manual detection using my screen reader, but I also used Accessibility Insight for Web (Chromium based  extension). 
You can find the extension in the Chrome or Microsoft Edge Extension store or you can find more info in the given URL.
https://accessibilityinsights.io/docs/web/overview/

I used the "Fast pass" option to generate the accessibility issues. Please find as a attachment.

Please note I have collected only from  two pages, but as I could confirm all of the form or dropdown or button has the same issue as any of the menues. 

1. Buttons do not have aria label text on the  pageNavigation page (left side)
   As I attached the result from the accessibility Insight for Web the following buttons:
 Assets, Assistance, Management, Tools, etc. buttons do not have aria labels or title. Also it worth to mentioned when I collapsed the navigation bar my screen reader can only read the following "Buttons without labell" on each of the mentioned buttons.

2. Collapse menu
   When I navigate to the "Collapse menu" and hit my space  or  enter key after the button has no aria label or title.
   3. Missing Aria labels for the keyboard shortcuts
   Somehow the Narrator and the NVDA could read some of the navigation menu keyboard shortcuts for example : Computer - Alt+o, Software - Alt+s, Home - Alt+1. However my JAWS could not read the keyboard  shortcuts for these menus. Interesting when I asked a person who has better eyes she said that when she tried with mouse to points to those menues which has  keyboard shortcut  the tooltip  did not show the shortcut for the menus.
   I suggest to add the aria label for the menues with the shortcuts as well  as for the tooltip.
   4. Form and drop down elements do not have  title and aria labels
   When I am adding a new computer  there are 9 dropdown list without title and aria labels. Also one other important thing with a keyboard there is no way to move when I open a dropdown list and it also worth to highlighted with screen reader nothing to read with it. It means I cannot hear anything when I am trying to moving between the listed elements inside te dropdown list. But this sympthom is also true for any of the similar dropdown lists inside the GLPI.

5. Find menu
   When I hit the find menu button there is a dropdown list appears and when I am trying to type any of the menu keyword with screen reader it is impossible to move as I cannot hear anything with my screen readers.

   6. Find menu results without keyboard shortcut help
   (This is just a suggestion to improve and teach the users to use the )menu(s) with keyboard.)
   When I am typing inside the find menu the result(s) do not show the keyboard shortcut. With this the users can easily can learn the keyboard shortcuts  for the submenus.

   I am attaching the result what I created with Accessibility Insight for Web Extension.
   Please see as an attachment.

   Please try to use the Accessibility Insight for Web  extension finding the accessibility bugs before the next release created for public.

   Please let me know if you need anything else from my side.

   Thanks for your support,
   Regards,
   Gabor

[FastPass_20251122_ComputerNewitemGLPI.html](https://github.com/user-attachments/files/23696523/FastPass_20251122_ComputerNewitemGLPI.html)
[FastPass_20251122_StandardinterfaceGLP.html](https://github.com/user-attachments/files/23696522/FastPass_20251122_StandardinterfaceGLP.html)

**Comment by cconard96** [comment]
Thank you for your report.

> Please try to use the Accessibility Insight for Web extension finding the accessibility bugs before the next release created for public.

This is not a task that can be completed in just a few days for the next bugfix release. Many of the accessibility issues lie with external libraries and even the issues that are with GLPI's own code are not always so easy to fix.

There is an issue open currently asking to work towards WCAG/RGAA compliance (#5996), but this task would likely be done as part of a larger UI/UX effort.

I'll leave this open in case there are quick fixes though that can be done for the specific issues reported here.

**Comment by citygabor** [comment]
Hello,

Thanks for your quick response,

Till that time the fix can be done in this issue: https://github.com/glpi-project/glpi/issues/5996. You may add the keyboard shortcuts as a tooltip and for the find menu results as well. The tooltip will not help for screen reader users, but may help for who has no problem with her/his eyes. In the Find menu the keyboard shortcut  key combination may added for the find menu results. like: Computer, Alt+O, Software, Alt+S, etc.

These request may not related for accessibility, but help for those who would like to use GLPI with keyboard shortcuts.

Thanks for your support,
Regards,
G
